From d191796e796946ff51d8e96d3fd1738f472813c4 Mon Sep 17 00:00:00 2001
From: Marek 'marx' Grac <mgrac@redhat.com>
Date: Thu, 19 Mar 2015 21:28:22 +0100
Subject: [PATCH 2/4] fence_ipmilan: Un-set default cipher

Previously, the default cipher was left to ipmitool, in one of latest releases we have changed it
to cipher=0. This commit removes this default and uses --cipher for ipmitool only when explicitly
set by user.

Resolves: rhbz#1203877
---
 fence/agents/ipmilan/fence_ipmilan.py | 4 ++--
 tests/data/metadata/fence_idrac.xml   | 2 +-
 tests/data/metadata/fence_ilo3.xml    | 2 +-
 tests/data/metadata/fence_ilo4.xml    | 2 +-
 tests/data/metadata/fence_imm.xml     | 2 +-
 tests/data/metadata/fence_ipmilan.xml | 2 +-
 6 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/fence/agents/ipmilan/fence_ipmilan.py b/fence/agents/ipmilan/fence_ipmilan.py
index 940b4be..30cfca9 100644
--- a/fence/agents/ipmilan/fence_ipmilan.py
+++ b/fence/agents/ipmilan/fence_ipmilan.py
@@ -53,7 +53,8 @@ def create_command(options, action):
 		cmd += " -P ''"
 
 	# --cipher / -C
-	cmd += " -C " + options["--cipher"]
+	if "--cipher" in options:
+		cmd += " -C " + options["--cipher"]
 
 	# --port / -n
 	if options.has_key("--ipport"):
@@ -96,7 +97,6 @@ def define_new_opts():
 		"help" : "-C, --cipher=[cipher]          Ciphersuite to use (same as ipmitool -C parameter)",
 		"required" : "0",
 		"shortdesc" : "Ciphersuite to use (same as ipmitool -C parameter)",
-		"default" : "0",
 		"order": 1
 	}
 	all_opt["privlvl"] = {
diff --git a/tests/data/metadata/fence_idrac.xml b/tests/data/metadata/fence_idrac.xml
index 1abdbd4..1455897 100644
--- a/tests/data/metadata/fence_idrac.xml
+++ b/tests/data/metadata/fence_idrac.xml
@@ -23,7 +23,7 @@
 	</parameter>
 	<parameter name="cipher" unique="0" required="0">
 		<getopt mixed="-C, --cipher=[cipher]" />
-		<content type="string" default="0"  />
+		<content type="string"  />
 		<shortdesc lang="en">Ciphersuite to use (same as ipmitool -C parameter)</shortdesc>
 	</parameter>
 	<parameter name="inet4_only" unique="0" required="0">
diff --git a/tests/data/metadata/fence_ilo3.xml b/tests/data/metadata/fence_ilo3.xml
index 6074a3c..02c9183 100644
--- a/tests/data/metadata/fence_ilo3.xml
+++ b/tests/data/metadata/fence_ilo3.xml
@@ -23,7 +23,7 @@
 	</parameter>
 	<parameter name="cipher" unique="0" required="0">
 		<getopt mixed="-C, --cipher=[cipher]" />
-		<content type="string" default="0"  />
+		<content type="string"  />
 		<shortdesc lang="en">Ciphersuite to use (same as ipmitool -C parameter)</shortdesc>
 	</parameter>
 	<parameter name="inet4_only" unique="0" required="0">
diff --git a/tests/data/metadata/fence_ilo4.xml b/tests/data/metadata/fence_ilo4.xml
index 234c9c6..28a77da 100644
--- a/tests/data/metadata/fence_ilo4.xml
+++ b/tests/data/metadata/fence_ilo4.xml
@@ -23,7 +23,7 @@
 	</parameter>
 	<parameter name="cipher" unique="0" required="0">
 		<getopt mixed="-C, --cipher=[cipher]" />
-		<content type="string" default="0"  />
+		<content type="string"  />
 		<shortdesc lang="en">Ciphersuite to use (same as ipmitool -C parameter)</shortdesc>
 	</parameter>
 	<parameter name="inet4_only" unique="0" required="0">
diff --git a/tests/data/metadata/fence_imm.xml b/tests/data/metadata/fence_imm.xml
index 33a75ec..3eb0932 100644
--- a/tests/data/metadata/fence_imm.xml
+++ b/tests/data/metadata/fence_imm.xml
@@ -23,7 +23,7 @@
 	</parameter>
 	<parameter name="cipher" unique="0" required="0">
 		<getopt mixed="-C, --cipher=[cipher]" />
-		<content type="string" default="0"  />
+		<content type="string"  />
 		<shortdesc lang="en">Ciphersuite to use (same as ipmitool -C parameter)</shortdesc>
 	</parameter>
 	<parameter name="inet4_only" unique="0" required="0">
diff --git a/tests/data/metadata/fence_ipmilan.xml b/tests/data/metadata/fence_ipmilan.xml
index 36027bd..34bc669 100644
--- a/tests/data/metadata/fence_ipmilan.xml
+++ b/tests/data/metadata/fence_ipmilan.xml
@@ -23,7 +23,7 @@
 	</parameter>
 	<parameter name="cipher" unique="0" required="0">
 		<getopt mixed="-C, --cipher=[cipher]" />
-		<content type="string" default="0"  />
+		<content type="string"  />
 		<shortdesc lang="en">Ciphersuite to use (same as ipmitool -C parameter)</shortdesc>
 	</parameter>
 	<parameter name="inet4_only" unique="0" required="0">
-- 
1.9.3

